conan/tools/microsoft/__init__.py

```diff
@@ -2,3 +2,17 @@
 from .toolchain import MSBuildToolchain
 from .msbuild import MSBuild
 from .msbuilddeps import MSBuildDeps
+
+
+def msvc_runtime_flag(conanfile):
+    settings = conanfile.settings
+    compiler = settings.get_safe("compiler")
+    runtime = settings.get_safe("compiler.runtime")
+    if compiler == "Visual Studio":
+        return runtime
+    if compiler == "msvc":
+        runtime_type = settings.get_safe("compiler.runtime_type")
+        if runtime == "static":
```

Comment:
this is a better implementation (from below):
```
        runtime = "MT" if runtime == "static" else "MD"
        if  runtime_type == "Debug":
            runtime = "{}d".format(runtime)
```
also, why not reuse?

Reply:
Changed.

Premature reuse is like premature optimization. It leads to wrong and obscure abstractions. I prefer stronger encapsulation, and knowing ``msvc_runtime_flag()`` is fully self-contained. Anyway, which other piece of code is mapping the new msvc settings to the runtime flag? I see it nowhere else, it is indeed a new functionality necessary for the new model.

Reply:
the code was from `def msvc_compatible(self):` (info.py).